src/main/java/killua/parser/Parser.java

```diff
@@ -0,0 +1,242 @@
+        Task task;
+
+        if (taskType == 'T') {
+            task = new Todo(argument);
+        } else if (taskType == 'D') {
+            String[] strs = argument.split("\\|", 2);
+            task = getDeadline(strs[0].strip(), strs[1].strip());
+        } else if (taskType == 'E') {
+            String[] strs = argument.split("\\|", 3);
+            task = getEvent(strs[0].strip(), strs[1].strip(), strs[2].strip());
+        } else {
+            throw new IllegalArgumentException("Unknown task type: " + taskType);
+        }
+
+        if (isDone) {
+            task.markAsDone();
+        }
+
+        return task;
+    }
+
+    /**
+     * Creates a Deadline task from the description and date string.
+     *
+     * @param description The description of the deadline.
+     * @param dateTimeString The date string for the deadline.
+     * @return The Deadline task created from the given description and date string.
+     * @throws IllegalArgumentException If the date format is invalid.
+     */
+    private static Task getDeadline(String description, String dateTimeString) {
+        try {
+            LocalDateTime dateTime = LocalDateTime.parse(dateTimeString, DATE_TIME_FORMATTER);
+            return new Deadline(description, dateTime);
+        } catch (DateTimeParseException e1) {
+            try {
+                LocalDate date = LocalDate.parse(dateTimeString, DATE_FORMATTER);
+                return new Deadline(description, date);
+            } catch (DateTimeParseException e2) {
+                throw new IllegalArgumentException("Invalid date format: " + dateTimeString);
+            }
+        }
+    }
+
+    /**
+     * Creates an Event task from the description and date strings.
+     *
+     * @param description The description of the event.
+     * @param dateTimeStringFrom The start date string of the event.
+     * @param dateTimeStringTo The end date string of the event.
+     * @return The Event task created from the given description and date strings.
+     * @throws IllegalArgumentException If the date format is invalid.
+     */
+    private static Task getEvent(String description, String dateTimeStringFrom, String dateTimeStringTo) {
+        try {
+            LocalDateTime fromDateTime = LocalDateTime.parse(dateTimeStringFrom, DATE_TIME_FORMATTER);
+            LocalDateTime toDateTime = LocalDateTime.parse(dateTimeStringTo, DATE_TIME_FORMATTER);
+            return new Event(description, fromDateTime, toDateTime);
+        } catch (DateTimeParseException e1) {
+            try {
+                LocalDate fromDate = LocalDate.parse(dateTimeStringFrom, DATE_FORMATTER);
+                LocalDate toDate = LocalDate.parse(dateTimeStringTo, DATE_FORMATTER);
+                return new Event(description, fromDate, toDate);
+            } catch (DateTimeParseException e2) {
+                throw new IllegalArgumentException("Invalid date format: from '" + dateTimeStringFrom + "' to '" + dateTimeStringTo + "'");
```

Comment:
I find that this line is too long and I noticed the same issue in several other places too. Perhaps you can wrap some parts to the next line?